scipy/spatial/tests/test_distance.py

```diff
@@ -305,23 +608,22 @@ def test_cdist_dtype_equivalence(self):
             for test in tests:
                 X1 = test[0]
                 try:
-                    y1 = pdist(X1, metric=metric)
+                    y1 = cdist(X1, metric=metric)
```

Comment:
good catch! My fault, copy and paste from pdist.
But you should also add also X2 I think as done in calling_convention:

      X1 = eo[eo_name][:, ::-1]
      X2 = eo[eo_name][:-3:2]